server/src/test/java/org/opensearch/action/admin/cluster/node/tasks/TaskTests.java

```diff
@@ -70,6 +72,11 @@ public void testTaskInfoToString() {
         assertEquals(((Number) map.get("start_time_in_millis")).longValue(), startTime);
         assertEquals(((Number) map.get("running_time_in_nanos")).longValue(), runningTime);
         assertEquals(map.get("cancellable"), cancellable);
+        if (cancellable) {
```

Comment:
Having `if` inside tests is an antipattern, it causes tests to start randomly failing. This test will randomly flip cancellable and not and sometimes test the functionality, and sometimes not. Split into three tests: the existing one that tests the kitchen sink, one that only tests cancellable and the other than only tests non-cancellable options.